Lib/ufoLib/glifLib.py

```diff
@@ -1187,21 +1194,25 @@ def _validateAndMassagePointStructures(contour, pointAttributes, openContourOffC
 		if element.tag != "point":
 			raise GlifLibError("Unknown child element (%s) of contour element." % element.tag)
 		# unknown attributes
-		unknownAttributes = [attr for attr in list(element.attrib.keys()) if attr not in pointAttributes]
-		if unknownAttributes:
-			raise GlifLibError("Unknown attributes in point element.")
+		for attr in element.attrib.keys():
+			if attr not in pointAttributes:
+				raise GlifLibError("Unknown attribute in point element: %s" % attr)
 		# search for unknown children
 		if len(element):
 			raise GlifLibError("Unknown child elements in point element.")
 		# x and y are required
-		x = element.get("x")
-		y = element.get("y")
-		if x is None:
-			raise GlifLibError("Required x attribute is missing in point element.")
-		if y is None:
-			raise GlifLibError("Required y attribute is missing in point element.")
-		element.attrib["x"] = _number(x)
-		element.attrib["y"] = _number(y)
+		for attr in ("x", "y"):
+			value = element.get(attr)
```

Comment:
maybe add a comment to explain why you deemed necessary to "inline" the `_number` function in this specific place